dnsmasq/Dockerfile

```diff
@@ -16,4 +16,5 @@ FROM alpine:3.4
 MAINTAINER Girish Kalele <gkalele@google.com>
 RUN apk --no-cache add dnsmasq
 COPY dnsmasq.conf /etc/dnsmasq.conf
+STOPSIGNAL SIGCONT
```

Comment:
I think you are correct. By using this preStop hook, we don't need to separately configure kubedns and dnsmasq to support graceful termination...